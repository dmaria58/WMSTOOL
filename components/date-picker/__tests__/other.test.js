import React from 'react';
import { mount, render } from 'enzyme';
import moment from 'moment';
import { MonthPicker, WeekPicker } from '../';

describe('MonthPicker and WeekPicker', () => {
  it('render MonthPicker', () => {
    const birthday = moment('2000-01-01', 'YYYY-MM-DD').locale('zh-cn');
    const wrapper = mount(
      <MonthPicker open />
    );
    wrapper.setProps({ value: birthday });
    expect(render(wrapper.find('Trigger').instance().getComponent())).toMatchSnapshot();
  });

  it('render WeekPicker', () => {
    const birthday = moment('2000-01-01', 'YYYY-MM-DD').locale('zh-cn');
    const wrapper = mount(
      <WeekPicker open />
    );
    wrapper.setProps({ value: birthday });
    expect(render(wrapper.find('Trigger').instance().getComponent())).toMatchSnapshot();
  });
});
