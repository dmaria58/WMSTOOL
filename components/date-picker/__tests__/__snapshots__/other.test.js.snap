// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthPicker and WeekPicker render MonthPicker 1`] = `
<div>
  <div
    class="ant-calendar-picker-container  ant-calendar-picker-container-placement-bottomLeft "
  >
    <div
      class="ant-calendar ant-calendar-month ant-calendar-month-calendar"
      tabindex="0"
    >
      <div
        class="ant-calendar-month-calendar-content"
      >
        <div
          class="ant-calendar-month-header-wrap"
        >
          <div
            class="ant-calendar-header"
          >
            <div
              style="position:relative"
            >
              <a
                class="ant-calendar-prev-year-btn"
                role="button"
                title="Last year (Control + left)"
              />
              <a
                class="ant-calendar-prev-month-btn"
                role="button"
                title="Previous month (PageUp)"
              />
              <span
                class="ant-calendar-my-select"
              >
                <a
                  class="ant-calendar-month-select"
                  role="button"
                  title="Choose a month"
                >
                  1月
                </a>
                <a
                  class="ant-calendar-year-select"
                  role="button"
                  title="Choose a year"
                >
                  2000
                </a>
              </span>
              <a
                class="ant-calendar-next-month-btn"
                title="Next month (PageDown)"
              />
              <a
                class="ant-calendar-next-year-btn"
                title="Next year (Control + right)"
              />
            </div>
            <div
              class="ant-calendar-month-panel"
            >
              <div>
                <div
                  class="ant-calendar-month-panel-header"
                >
                  <a
                    class="ant-calendar-month-panel-prev-year-btn"
                    role="button"
                    title="Last year (Control + left)"
                  />
                  <a
                    class="ant-calendar-month-panel-year-select"
                    role="button"
                    title="Choose a year"
                  >
                    <span
                      class="ant-calendar-month-panel-year-select-content"
                    >
                      2000
                    </span>
                    <span
                      class="ant-calendar-month-panel-year-select-arrow"
                    >
                      x
                    </span>
                  </a>
                  <a
                    class="ant-calendar-month-panel-next-year-btn"
                    role="button"
                    title="Next year (Control + right)"
                  />
                </div>
                <div
                  class="ant-calendar-month-panel-body"
                >
                  <table
                    cellspacing="0"
                    class="ant-calendar-month-panel-table"
                    role="grid"
                  >
                    <tbody
                      class="ant-calendar-month-panel-tbody"
                    >
                      <tr
                        role="row"
                      >
                        <td
                          class="ant-calendar-month-panel-cell ant-calendar-month-panel-selected-cell"
                          role="gridcell"
                          title="一月"
                        >
                          <a
                            class="ant-calendar-month-panel-month"
                          >
                            一月
                          </a>
                        </td>
                        <td
                          class="ant-calendar-month-panel-cell"
                          role="gridcell"
                          title="二月"
                        >
                          <a
                            class="ant-calendar-month-panel-month"
                          >
                            二月
                          </a>
                        </td>
                        <td
                          class="ant-calendar-month-panel-cell"
                          role="gridcell"
                          title="三月"
                        >
                          <a
                            class="ant-calendar-month-panel-month"
                          >
                            三月
                          </a>
                        </td>
                      </tr>
                      <tr
                        role="row"
                      >
                        <td
                          class="ant-calendar-month-panel-cell"
                          role="gridcell"
                          title="四月"
                        >
                          <a
                            class="ant-calendar-month-panel-month"
                          >
                            四月
                          </a>
                        </td>
                        <td
                          class="ant-calendar-month-panel-cell"
                          role="gridcell"
                          title="五月"
                        >
                          <a
                            class="ant-calendar-month-panel-month"
                          >
                            五月
                          </a>
                        </td>
                        <td
                          class="ant-calendar-month-panel-cell"
                          role="gridcell"
                          title="六月"
                        >
                          <a
                            class="ant-calendar-month-panel-month"
                          >
                            六月
                          </a>
                        </td>
                      </tr>
                      <tr
                        role="row"
                      >
                        <td
                          class="ant-calendar-month-panel-cell"
                          role="gridcell"
                          title="七月"
                        >
                          <a
                            class="ant-calendar-month-panel-month"
                          >
                            七月
                          </a>
                        </td>
                        <td
                          class="ant-calendar-month-panel-cell"
                          role="gridcell"
                          title="八月"
                        >
                          <a
                            class="ant-calendar-month-panel-month"
                          >
                            八月
                          </a>
                        </td>
                        <td
                          class="ant-calendar-month-panel-cell"
                          role="gridcell"
                          title="九月"
                        >
                          <a
                            class="ant-calendar-month-panel-month"
                          >
                            九月
                          </a>
                        </td>
                      </tr>
                      <tr
                        role="row"
                      >
                        <td
                          class="ant-calendar-month-panel-cell"
                          role="gridcell"
                          title="十月"
                        >
                          <a
                            class="ant-calendar-month-panel-month"
                          >
                            十月
                          </a>
                        </td>
                        <td
                          class="ant-calendar-month-panel-cell"
                          role="gridcell"
                          title="十一月"
                        >
                          <a
                            class="ant-calendar-month-panel-month"
                          >
                            十一月
                          </a>
                        </td>
                        <td
                          class="ant-calendar-month-panel-cell"
                          role="gridcell"
                          title="十二月"
                        >
                          <a
                            class="ant-calendar-month-panel-month"
                          >
                            十二月
                          </a>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MonthPicker and WeekPicker render WeekPicker 1`] = `
<div>
  <div
    class="ant-calendar-picker-container  ant-calendar-picker-container-placement-bottomLeft "
  >
    <div
      class="ant-calendar ant-calendar-week-number"
      tabindex="0"
    >
      <div
        class="ant-calendar-panel"
      >
        <div
          class="ant-calendar-date-panel"
        >
          <div
            class="ant-calendar-header"
          >
            <div
              style="position:relative"
            >
              <a
                class="ant-calendar-prev-year-btn"
                role="button"
                title="Last year (Control + left)"
              />
              <a
                class="ant-calendar-prev-month-btn"
                role="button"
                title="Previous month (PageUp)"
              />
              <span
                class="ant-calendar-my-select"
              >
                <a
                  class="ant-calendar-month-select"
                  role="button"
                  title="Choose a month"
                >
                  1月
                </a>
                <a
                  class="ant-calendar-year-select"
                  role="button"
                  title="Choose a year"
                >
                  2000
                </a>
              </span>
              <a
                class="ant-calendar-next-month-btn"
                title="Next month (PageDown)"
              />
              <a
                class="ant-calendar-next-year-btn"
                title="Next year (Control + right)"
              />
            </div>
          </div>
          <div
            class="ant-calendar-body"
          >
            <table
              cellspacing="0"
              class="ant-calendar-table"
              role="grid"
            >
              <thead>
                <tr
                  role="row"
                >
                  <th
                    class="ant-calendar-column-header ant-calendar-week-number-header"
                    role="columnheader"
                  >
                    <span
                      class="ant-calendar-column-header-inner"
                    >
                      x
                    </span>
                  </th>
                  <th
                    class="ant-calendar-column-header"
                    role="columnheader"
                    title="周一"
                  >
                    <span
                      class="ant-calendar-column-header-inner"
                    >
                      一
                    </span>
                  </th>
                  <th
                    class="ant-calendar-column-header"
                    role="columnheader"
                    title="周二"
                  >
                    <span
                      class="ant-calendar-column-header-inner"
                    >
                      二
                    </span>
                  </th>
                  <th
                    class="ant-calendar-column-header"
                    role="columnheader"
                    title="周三"
                  >
                    <span
                      class="ant-calendar-column-header-inner"
                    >
                      三
                    </span>
                  </th>
                  <th
                    class="ant-calendar-column-header"
                    role="columnheader"
                    title="周四"
                  >
                    <span
                      class="ant-calendar-column-header-inner"
                    >
                      四
                    </span>
                  </th>
                  <th
                    class="ant-calendar-column-header"
                    role="columnheader"
                    title="周五"
                  >
                    <span
                      class="ant-calendar-column-header-inner"
                    >
                      五
                    </span>
                  </th>
                  <th
                    class="ant-calendar-column-header"
                    role="columnheader"
                    title="周六"
                  >
                    <span
                      class="ant-calendar-column-header-inner"
                    >
                      六
                    </span>
                  </th>
                  <th
                    class="ant-calendar-column-header"
                    role="columnheader"
                    title="周日"
                  >
                    <span
                      class="ant-calendar-column-header-inner"
                    >
                      日
                    </span>
                  </th>
                </tr>
              </thead>
              <tbody
                class="ant-calendar-tbody"
              >
                <tr
                  class="ant-calendar-active-week"
                  role="row"
                >
                  <td
                    class="ant-calendar-week-number-cell"
                    role="gridcell"
                  >
                    52
                  </td>
                  <td
                    class="ant-calendar-cell ant-calendar-last-month-cell"
                    role="gridcell"
                    title="1999年12月27日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      27
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell ant-calendar-last-month-cell"
                    role="gridcell"
                    title="1999年12月28日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      28
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell ant-calendar-last-month-cell"
                    role="gridcell"
                    title="1999年12月29日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      29
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell ant-calendar-last-month-cell"
                    role="gridcell"
                    title="1999年12月30日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      30
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell ant-calendar-last-month-cell"
                    role="gridcell"
                    title="1999年12月31日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      31
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell ant-calendar-selected-date ant-calendar-selected-day"
                    role="gridcell"
                    title="2000年1月1日"
                  >
                    <div
                      class="ant-calendar-selected-day"
                    >
                      <div
                        class="ant-calendar-date"
                      >
                        1
                      </div>
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="2000年1月2日"
                  >
                    <div
                      class="ant-calendar-selected-day"
                    >
                      <div
                        class="ant-calendar-date"
                      >
                        2
                      </div>
                    </div>
                  </td>
                </tr>
                <tr
                  class=""
                  role="row"
                >
                  <td
                    class="ant-calendar-week-number-cell"
                    role="gridcell"
                  >
                    1
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="2000年1月3日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      3
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="2000年1月4日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      4
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="2000年1月5日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      5
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="2000年1月6日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      6
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="2000年1月7日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      7
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="2000年1月8日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      8
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="2000年1月9日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      9
                    </div>
                  </td>
                </tr>
                <tr
                  class=""
                  role="row"
                >
                  <td
                    class="ant-calendar-week-number-cell"
                    role="gridcell"
                  >
                    2
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="2000年1月10日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      10
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="2000年1月11日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      11
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="2000年1月12日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      12
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="2000年1月13日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      13
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="2000年1月14日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      14
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="2000年1月15日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      15
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="2000年1月16日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      16
                    </div>
                  </td>
                </tr>
                <tr
                  class=""
                  role="row"
                >
                  <td
                    class="ant-calendar-week-number-cell"
                    role="gridcell"
                  >
                    3
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="2000年1月17日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      17
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="2000年1月18日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      18
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="2000年1月19日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      19
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="2000年1月20日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      20
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="2000年1月21日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      21
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="2000年1月22日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      22
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="2000年1月23日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      23
                    </div>
                  </td>
                </tr>
                <tr
                  class=""
                  role="row"
                >
                  <td
                    class="ant-calendar-week-number-cell"
                    role="gridcell"
                  >
                    4
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="2000年1月24日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      24
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="2000年1月25日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      25
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="2000年1月26日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      26
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="2000年1月27日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      27
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="2000年1月28日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      28
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="2000年1月29日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      29
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="2000年1月30日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      30
                    </div>
                  </td>
                </tr>
                <tr
                  class=""
                  role="row"
                >
                  <td
                    class="ant-calendar-week-number-cell"
                    role="gridcell"
                  >
                    5
                  </td>
                  <td
                    class="ant-calendar-cell"
                    role="gridcell"
                    title="2000年1月31日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      31
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell ant-calendar-next-month-btn-day"
                    role="gridcell"
                    title="2000年2月1日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      1
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell ant-calendar-next-month-btn-day"
                    role="gridcell"
                    title="2000年2月2日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      2
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell ant-calendar-next-month-btn-day"
                    role="gridcell"
                    title="2000年2月3日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      3
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell ant-calendar-next-month-btn-day"
                    role="gridcell"
                    title="2000年2月4日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      4
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell ant-calendar-next-month-btn-day"
                    role="gridcell"
                    title="2000年2月5日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      5
                    </div>
                  </td>
                  <td
                    class="ant-calendar-cell ant-calendar-next-month-btn-day"
                    role="gridcell"
                    title="2000年2月6日"
                  >
                    <div
                      class="ant-calendar-calendar-date"
                    >
                      6
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
